### PR TITLE
sscep: add package

### PR DESCRIPTION
**Pull request by johnfzc**
Maintainer: @johnfzc
Compile tested: rampis, glinet_gl-mt300n-v2, 22.03.3
Run tested: rampis, glinet_gl-mt300n-v2, 22.03.3

Description: SSCEP client from certnanny, it was previously packaged in OpenWrt and removed at some point, I've been maintaining it outside of the tree since.

### CODE REVIEW

File: net/sscep/Makefile
Comment (BKPepe) [inline net/sscep/Makefile:6]: Remove it or assign it to yourself.
Check: https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#makefile-contents-should-contain

File: net/sscep/Makefile
Comment (BKPepe) [inline net/sscep/Makefile:14]: This isn't good, because in dl folder, the tarball will be downloaded as v0.10.0 and it would be hard to know for which sw it was downloaded. It should be packageName-version.

File: net/sscep/Makefile
Comment (BKPepe) [inline net/sscep/Makefile:18]: Based on this https://github.com/certnanny/sscep/blob/master/COPYING, I see BSD-3-Clause, OpenSSL and others. 
Anyway, you should add the LICENSE file.

File: net/sscep/Makefile
Comment (BKPepe) [inline net/sscep/Makefile:47]: Remove it.

**Review by BKPepe — CHANGES_REQUESTED** [review]
1. I wonder, why it was removed back?
   - Why do we need it to add it back?
 
 2. It would be nice to have ``test.sh`` file, which ensures that this package is run tested by CI.
 See: https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#continuous-integration
 
 3. Package description is missing.
 
 4. I think Prepare,Configure and Compile stages can be improved a lot.

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:41]: It's better to switch to use CMake.

**Comment by johnfzc** [comment]
> * I wonder, why it was removed back?

I am not certain, I suspect it's maintainer may have abandoned it.

I can find it here: https://git.openwrt.org/?p=openwrt/svn-archive/packages.git;a=commit;h=b7f060b6ee758612a7a15ae26faf649269362667

And in oldpackages: https://github.com/mhei/openwrt-oldpackages/commit/f8fd1c82f6ebd164272715d6012a5930ca714566

But I can't find a trace of it's removal in the current packages tree.

>   * Why do we need it to add it back?

OpenWrt includes another [SCEP client](https://openwrt.org/packages/pkgdata/strongswan-scepclient), it doesn't work with my SCEP server.

> * It would be nice to have `test.sh` file, which ensures that this package is run tested by CI.
>   See: https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#continuous-integration

I'll add one and address the other concerns and comments if the package will be accepted once they are addressed. Please let me know what you think of my response about the removal and why to add it back first.

Thanks very much for taking the time to review this PR.

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:14]: This name is based on the naming used by the upstream package; here's the release page:

https://github.com/certnanny/sscep/releases/tag/v0.10.0

And the tar ball that's linked from it:

https://github.com/certnanny/sscep/archive/refs/tags/v0.10.0.tar.gz

Is there a way that I can download it to a different name as the naming. while mandatory for the download,  is highly confusing in the dl folder?

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:41]: How can I do this? Can you please point me to an example package?

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:7]: @BKPepe this is the last remaining issue raised in the review, I've taken a look at the [documentation](https://openwrt.org/docs/guide-developer/packages) for the variables in a package Makefile and don't see how I can download the package to a different local name than the remote file name reflected in the `PKG_SOURCE`.

Please let me know if you see a way this can be solved. Thanks again!

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:8]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/certnanny/sscep/tar.gz/v$(PKG_VERSION)?
```

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:12]:
```suggestion
PKG_LICENSE:=BSD-3-Clause
```

from the COPYING file, looks like it's BSD 3-Clause

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:29]:
```suggestion
  It can be used to retrieve CA certificates and enroll devices with
  a compatible SCEP server.
```

wrap new line for every 70 chars

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:39]:
```suggestion
```

it's for the autotools, can be removed for cmake

File: net/sscep/Makefile
Comment (1715173329) [inline net/sscep/Makefile:50]:
```suggestion

	$(INSTALL_DIR) $(1)/usr/bin
	$(INSTALL_BIN) $(PKG_BUILD_DIR)/sscep $(1)/usr/bin/sscep
	$(INSTALL_BIN) $(PKG_BUILD_DIR)/mkrequest $(1)/usr/bin/mkrequest
```

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: net/sscep/Makefile
Comment (jefferyto) [reply net/sscep/Makefile:7]: From the link the file name looks like it will be v0.10.0.tar.gz, but if you click the link the download name is actually sscep-0.10.0.tar.gz.

File: net/sscep/Makefile
Comment (jefferyto) [reply net/sscep/Makefile:12]: @BKPepe was right (see https://github.com/openwrt/packages/pull/20683#discussion_r1140684101), the file contains multiple licenses.

I think it would be most appropriate to list all of the licenses (space-separated) in `PKG_LICENSE`.

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:7]: That's very interesting @jefferyto, but I don't understand how it works. The actual URI used for the download is https://github.com/certnanny/sscep/archive/refs/tags/v0.10.0.tar.gz and the browser appears to be renaming the file.

Can you give me a raw uri that works and contains the sscep file name? I tried looking at the network tab in my browser to validate what was happening but couldn't find a URI or understand how the apparent rename is occurring.

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:12]: Ok, that makes sense, I'll do that.

**Comment by johnfzc** [comment]
I haven't been able to work on this yet but will address all of the provided feedback. I also had tested this new version but apparently some remnants of my previous build made it (appear to?) work as the CI has all failed. I'll do a cleaner job of testing it to figure out why it's no longer building and fix that as well. If you see an obvious reason for the build failures please let me know.

Thanks for the comments that will make this package better and clean, really appreciate it.

File: net/sscep/Makefile
Comment (jefferyto) [reply net/sscep/Makefile:7]: It's the `Content-Disposition: attachment; filename=sscep-0.10.0.tar.gz` header in the final 200 response, after all the redirections.

You can also use codeload as @1715173329 [suggested](https://github.com/openwrt/packages/pull/20683#discussion_r1148317548).

There are many packages here that download from GitHub, you can check other package makefiles to see how they do it.

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:12]: I wasn't able to add the OpenOSP license, I've put in the two that are defined in SPDX.

File: net/sscep/Makefile
Comment (johnfzc) [reply net/sscep/Makefile:50]: Sorry, this request isn't clear to me. I tried to switch from tabs to spaces as I thought that was what you were suggesting but it prevented the Makefile from working. I looked at the msgpack-c Makefile and it appears to follow the same tab and space usage as is in my Makefile.

Comment (1715173329) [reply net/sscep/Makefile]:
```patch
--- a/net/sscep/Makefile
+++ b/net/sscep/Makefile
@@ -37,6 +37,7 @@ endef
 define Package/sscep/install
 	$(INSTALL_DIR) $(1)/etc
 	$(INSTALL_CONF) $(PKG_BUILD_DIR)/sscep.conf $(1)/etc/
+
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/sscep $(1)/usr/bin/sscep
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/mkrequest $(1)/usr/bin/mkrequest
```

**Review by 1715173329 — APPROVED** [review]
lgtm from my side, but it would be nice to have a init script.

**Comment by johnfzc** [comment]
>it would be nice to have a init script.

The sscep binary doesn't provide a running service. It's a client, like curl for http or an ftp client. You use it interactively, to request a certificate from a server on an on demand basis.

**Comment by jefferyto** [comment]
The first license in COPYING is BSD-2-Clause not BSD-3-Clause (you can tell by the [number of clauses](https://github.com/certnanny/sscep/blob/v0.10.0/COPYING#L8-L13)), and the OpenOSP license is the [Vovida Software License v1.0](https://spdx.org/licenses/VSL-1.0.html) (also [named in the file](https://github.com/certnanny/sscep/blob/v0.10.0/COPYING#L148)).

**Comment by johnfzc** [comment]
I can create a PR to clean that up, @jefferyto can you please confirm the line should be:

`PKG_LICENSE:=BSD-2-Clause OpenSSL VSL-1.0`

**Comment by jefferyto** [comment]
> I can create a PR to clean that up, @jefferyto can you please confirm the line should be:
> 
> `PKG_LICENSE:=BSD-2-Clause OpenSSL VSL-1.0`

Yes that looks correct - thanks :pray:

**Comment by johnfzc** [comment]
@jefferyto I've fixed up the license line and created PR #20846 please take a look at it and approve it or let me know if there are issues.

**Pull request merged**
